src/components/entity/state-badge.ts

```diff
@@ -79,6 +79,12 @@ export class StateBadge extends LitElement {
       </div>`;
     }
 
+    for (const _cls of ["has-image", "has-media-image", "has-no-radius"]) {
+      if (this.classList.contains(_cls)) this.classList.remove(_cls);
+    }
+    const cls = this.getClass();
+    if (cls) this.classList.add(cls);
```

Comment:
A nicer implementation would be to return a classMap from `this.getClass()` with true/false for each class and add or remove classes based on that. This way all the related classes will be defined in only one place.

Reply:
I think I have done it...